entity-view/testsuite/src/test/java/com/blazebit/persistence/view/testsuite/cte/model/PersonWithPartnerDocumentFullAged.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2014 - 2019 Blazebit.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.blazebit.persistence.view.testsuite.cte.model;
+
+import com.blazebit.persistence.testsuite.entity.Person;
+import com.blazebit.persistence.view.EntityView;
+import com.blazebit.persistence.view.EntityViewInheritance;
+import com.blazebit.persistence.view.EntityViewInheritanceMapping;
+import com.blazebit.persistence.view.testsuite.inheritance.subview.correlated.model.PersonBaseView;
+
+@EntityView(Person.class)
+@EntityViewInheritanceMapping("age < 18")
+public interface PersonWithPartnerDocumentFullAged extends PersonWithPartnerDocument {
```

Comment:
You will have to add a test that also adds that entity view to the configuration. Regardless of that, you might need to add a `@With` with a `CTEProvider`. Your tests don't have to check end-to-end that it works, just check that the metamodel contains an instance of the CTEProvider.

Reply:
> […] just check that the metamodel contains an instance of the CTEProvider.

Cool! Can I check that from `evm.getMetamodel()`?

Reply:
Yeah, you added the method to `ManagedViewType` in this PR, so you can ;)